src/docfx/build/toc/TableOfContentsMap.cs

```diff
@@ -58,49 +62,69 @@ public Document GetNearestToc(Document file)
             var filteredTocFiles = hasReferencedToc ? (IEnumerable<Document>)referencedTocFiles : _tocs;
```

Comment:
>(IEnumerable<Document>)referencedTocFiles [](start = 54, length = 41)

how about using this directly in the next check of hasReferencedToc to avoid extra check?